src/components/negotiations-panel.tsx

```diff
@@ -0,0 +1,271 @@
+        baseLabelText += ']';
+        baseLabelText += determineStatusText(baseStatus, conflict.resolved);
+
+        let baseStyle = determineStyle(baseStatus, conflict.resolved);
+
+        let terminalId = version + '.' + participantId + '.terminal'; //terminal node
+        let baseId = version + '.' + participantId + '.base'; //base ID
+        tableDom.push(
+          <TreeItem
+            data-component="TreeItem"
+            nodeId={terminalId}
+            key={terminalId}
+            classes={{ label: terminalStyle, selected: terminalStyle }}
+            label={terminalLabelText}
+          >
+            <TreeItem
+              data-component="TreeItem"
+              nodeId={baseId}
+              key={baseId}
+              classes={{ label: baseStyle, selected: baseStyle }}
+              label={baseLabelText}
+            />
+          </TreeItem>,
+        );
+
+        let terminalParams = {
+          map_name: curLevelName,
+          conflict_version: parseInt(version),
+          sequence: terminalStatus.sequence,
+        };
+        nodeidToParameters.set(terminalId, terminalParams);
+
+        let baseParams = {
+          map_name: curLevelName,
+          conflict_version: parseInt(version),
+          sequence: baseStatus.sequence,
+        };
+        nodeidToParameters.set(baseId, baseParams);
+      } else {
+        //single node
+        let baseStatus = statusData.base;
+        let labelText = participantName + determineStatusText(baseStatus, conflict.resolved);
+        let style = determineStyle(baseStatus, conflict.resolved);
+        let nodeId = version + '.' + participantId + '.base'; //base ID
+        tableDom.push(
+          <TreeItem
+            data-component="TreeItem"
+            nodeId={nodeId}
+            key={nodeId}
+            classes={{ label: style, selected: style }}
+            label={labelText}
+          />,
+        );
+
+        let baseParams = {
+          map_name: curLevelName,
+          conflict_version: parseInt(version),
+          sequence: baseStatus.sequence,
+        };
+        nodeidToParameters.set(nodeId, baseParams);
+      }
+    }
+
+    let nodeIdBase = 'conflict' + version;
+    return (
+      <TreeItem
+        data-component="TreeItem"
+        nodeId={nodeIdBase}
+        key={nodeIdBase}
+        classes={{ label: conflictStyle, selected: conflictStyle }}
+        label={conflictLabel}
+      >
+        {tableDom}
+      </TreeItem>
+    );
+  };
+
+  let negotiationContents: JSX.Element[] = [];
+  if (conflicts) {
+    let reversedConflicts = Object.keys(conflicts).reverse();
+    reversedConflicts.forEach(version => {
+      const conflict = conflicts[version];
+      let contents = renderNegotiations(version, conflict);
+      negotiationContents.push(contents);
+    });
+  } else {
+    console.log('prop negotationstatus empty');
+  }
+
+  // action callbacks
+  const handleSelect = (event: React.ChangeEvent<{}>, nodeIds: string): void => {
```

Comment:
I think you will get rid of a lot of these codebot notifications by mocking the `handleSelect` on treeView. You can also simulate  a click, you can check for an example here: 
https://github.com/osrf/romi-dashboard/blob/master/src/components/schedule-visualizer/door/tests/door.test.tsx


```
test('Trigger click event', async () => {
  let clicked = false;
  const handleClick = () => {
    clicked = true;
  };
  const wrapper = mount(
    <svg>
      <DefaultDoor
        v1={[mainDoor.v1_x, mainDoor.v1_y]}
        v2={[mainDoor.v2_x, mainDoor.v2_y]}
        onClick={handleClick}
      />
    </svg>,
  );
  wrapper
    .find(DefaultDoor)
    .at(0)
    .find('g')
    .simulate('click');
  expect(clicked).toBe(true);

  wrapper.unmount();
});
```

Reply:
the e2e is testing for this. catering for this test would include moving a function and its dependencies to the dashboard/index so that i can test 3 loc so I'd say skip it.

Reply:
Hmm ... yeah, you're right, we could let the e2e test that part like you're saying. Mmm .. It seems that all things are encapsulated in NegotiationsPanel, so it will be difficult to test internal methods of the component, so we could leave it as is and do black-box testing (as we have it now).